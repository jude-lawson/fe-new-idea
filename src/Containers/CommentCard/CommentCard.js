import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

export const CommentCard = (props) => {
  const {body} = props;
  return (
    <article>
      <button>REMOVE</button>
      <p>{body}</p>
      <button>COMMENT</button>
    </article>
  );
};

CommentCard.propTypes = {
  body: PropTypes.string
};