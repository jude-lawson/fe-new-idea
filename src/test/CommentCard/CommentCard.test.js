import React from 'react';
import { shallow } from 'enzyme';
import { CommentCard, mapStateToProps } from '../../Containers/CommentCard/CommentCard';

describe('CommentCard', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<CommentCard />);

    expect(wrapper).toMatchSnapshot();
  });
});