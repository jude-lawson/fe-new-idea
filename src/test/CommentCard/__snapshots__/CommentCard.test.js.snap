// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button>
          REMOVE
</button>,
        <p />,
        <button>
          COMMENT
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "REMOVE",
        },
        "ref": null,
        "rendered": "REMOVE",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "COMMENT",
        },
        "ref": null,
        "rendered": "COMMENT",
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button>
            REMOVE
</button>,
          <p />,
          <button>
            COMMENT
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "REMOVE",
          },
          "ref": null,
          "rendered": "REMOVE",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "COMMENT",
          },
          "ref": null,
          "rendered": "COMMENT",
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
